frontend/packages/ceph-storage-plugin/src/components/block-pool/block-pool-list-page.tsx

```diff
@@ -0,0 +1,252 @@
+      transforms: [sortable],
+      props: { className: blockPoolColumnInfo.name.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.status.title),
+      id: blockPoolColumnInfo.status.id,
+      props: { className: blockPoolColumnInfo.status.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.storageclasses.title),
+      id: blockPoolColumnInfo.storageclasses.id,
+      props: { className: blockPoolColumnInfo.storageclasses.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.replicas.title),
+      id: blockPoolColumnInfo.replicas.id,
+      props: { className: blockPoolColumnInfo.replicas.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.usedcapacity.title),
+      id: blockPoolColumnInfo.usedcapacity.id,
+      props: { className: blockPoolColumnInfo.usedcapacity.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.mirroringstatus.title),
+      id: blockPoolColumnInfo.mirroringstatus.id,
+      props: { className: blockPoolColumnInfo.mirroringstatus.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.mirroringhealth.title),
+      id: blockPoolColumnInfo.mirroringhealth.id,
+      props: { className: blockPoolColumnInfo.mirroringhealth.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.compressionstatus.title),
+      id: blockPoolColumnInfo.compressionstatus.id,
+      props: { className: blockPoolColumnInfo.compressionstatus.classes },
+    },
+    {
+      title: i18next.t(blockPoolColumnInfo.compressionsavings.title),
+      id: blockPoolColumnInfo.compressionsavings.id,
+      props: { className: blockPoolColumnInfo.compressionsavings.classes },
+    },
+    {
+      title: '',
+      props: { className: Kebab.columnClass },
+    },
+  ];
+};
+
+const getRows: RowFunction<K8sResourceKind> = (props) => <BlockPoolTableRow {...props} />;
+
+const BlockPoolTableRow: React.FC<RowFunctionArgs> = ({ obj, index, key, style, customData }) => {
+  const { t } = useTranslation();
+  const { name, namespace } = obj.metadata;
+  const replica: string = obj.spec?.replicated?.size;
+  const mirroringStatus: boolean = obj.spec?.mirroring?.enabled;
+  const mirroringHealth: string = obj.status?.mirroringStatus?.summary?.health;
+  const lastChecked: string = obj.status?.mirroringStatus?.lastChecked;
+  const compressionStatus: boolean = obj.spec?.compressionMode === COMPRESSION_ON;
+  const phase = obj.status?.phase;
+  const cephCluster: CephClusterKind = customData?.cephClusterObj;
+
+  // Hooks
+  const [scResources, scLoaded, scError] = customData?.storageClassResources;
+  const scNames: string[] = React.useMemo(
+    () => (scLoaded && !scError ? getScNamesUsingPool(scResources, name) : []),
+    [scResources, scLoaded, scError, name],
+  );
+
+  // Metrics
+  const queries = React.useMemo(() => (() => Object.values(getPerPoolQueryMap(name)))(), [name]);
+  const values = usePrometheusQueries(queries, parser as any);
```

Comment:
```suggestion
  const values = usePrometheusPoll(queries, parser as any);
```
`usePrometheusQueries` should only be used in Dashboards.

Reply:
Can you query all of them at once and pass it as customData to these rows. Don't want too many connections opening up from a single List Page.

Reply:
ack

Reply:
I feel bulk reading is dangerous when a number of pool count increases. I checked in other packages and no one doing bulk reading. I worried about in-memory.